src/DerConverter/Asn/DerAsnUtcTime.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DerConverter.Asn
+{
+    public class DerAsnUtcTime : DerAsnType
+    {
+        private readonly List<byte> _bytes = new List<byte>();
+
+        internal DerAsnUtcTime(Queue<byte> rawData)
+            : base(DerAsnTypeTag.UtcTime)
+        {
+            if (rawData == null) throw new ArgumentNullException(nameof(rawData));
+            _bytes.AddRange(rawData.DequeueAll());
+        }
+
+        public DerAsnUtcTime(DateTimeOffset time, bool includeSeconds = true)
+            : base(DerAsnTypeTag.UtcTime)
+        {
+            string timeAsString = null;
+
+            if (time.Offset == TimeSpan.Zero)
```

Comment:
First prepare the `timeZonePart` string, after that you only need one `if (includeSeconds)` as you can do `$"{time:yyMMddHHmmss}{timeZonePart}"`